example-react-app/src/app/App.tsx

```diff
@@ -26,35 +27,52 @@ const App = observer(() => {
 
   if (loginRequired) {
     return (
-      <Centered>
-        <Login />
-      </Centered>
+      <AppErrorBoundary>
+        <Centered>
+          <Login />
+        </Centered>
+      </AppErrorBoundary>
     );
   }
 
   return (
-    <Layout className={styles.mainLayout}>
-      <Layout.Header>
-        <AppHeader />
-      </Layout.Header>
-      <Layout className={styles.layoutContainer}>
-        <Layout.Sider
-          width={200}
-          breakpoint="sm"
-          collapsedWidth={0}
-          className={styles.layoutSider}
-        >
-          <AppMenu />
-        </Layout.Sider>
-
-        <Layout className={styles.layoutContent}>
-          <Layout.Content>
-            <Router global routes={routes} />
-          </Layout.Content>
+    <AppErrorBoundary>
+      <Layout className={styles.mainLayout}>
+        <Layout.Header>
+          <AppHeader />
+        </Layout.Header>
+        <Layout className={styles.layoutContainer}>
+          <Layout.Sider
+            width={200}
+            breakpoint="sm"
+            collapsedWidth={0}
+            className={styles.layoutSider}
+          >
+            <AppMenu />
+          </Layout.Sider>
+
+          <Layout className={styles.layoutContent}>
+            <Layout.Content>
+              <Router global routes={routes} />
+            </Layout.Content>
+          </Layout>
         </Layout>
       </Layout>
-    </Layout>
+    </AppErrorBoundary>
   );
 });
 
+const AppErrorBoundary = function(props) {
+  const intl = useIntl();
```

Comment:
I think we can get rid of this component, it should be possible to inject `intl` into a class component.